tutorials/appengine-pusher/index.md

```diff
@@ -0,0 +1,450 @@
+
+For more on publishing messages to multiple channels or batching multiple messages, refer to
+[this page](https://pusher.com/docs/server_api_guide/interact_rest_api#publishing-events).
+
+[embedmd]:# (java/src/main/java/com/example/appengine/pusher/SendMessageServlet.java /public class/ $)
+```java
+public class SendMessageServlet extends HttpServlet {
+
+  private Gson gson = new GsonBuilder().create();
+  private TypeReference<Map<String, String>> typeReference =
+      new TypeReference<Map<String, String>>() {};
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Parse POST request body received in the format:
+    // [{"message": "my-message", "socket_id": "1232.24", "channel": "presence-my-channel"}]
+
+    String body = CharStreams.readLines(request.getReader()).toString();
+    String json = body.replaceFirst("^\\[", "").replaceFirst("\\]$", "");
+    Map<String, String> data = gson.fromJson(json, typeReference.getType());
+    String message = data.get("message");
+    String socketId = data.get("socket_id");
+    String channelId = data.get("channel_id");
+
+    User user = UserServiceFactory.getUserService().getCurrentUser();
+    // User email prefix as display name for currently authenticated user
+    String displayName = user.getNickname().replaceFirst("@.*", "");
+
+    // Create a message including the user email prefix to display in the chat window
+    String taggedMessage = "<strong>&lt;" + displayName + "&gt;</strong> " + message;
+    Map<String, String> messageData = new HashMap<>();
+    messageData.put("message", taggedMessage);
+
+    // Send a message over the Pusher channel (maximum size of a message is 10KB)
+    Result result =
+        PusherService.getDefaultInstance()
+            .trigger(
+                channelId,
+                "new_message", // name of event
+                messageData,
+                socketId); // (Optional) Use client socket_id to exclude the sender from receiving the message
+
+    // result.getStatus() == SUCCESS indicates successful transmission
+    messageData.put("status", result.getStatus().name());
+
+    response.getWriter().println(gson.toJson(messageData));
+  }
+}
+```
+
+## Using Pusher on your client
+
+The following sections explain how to subscribe to Pusher channels on your client using the
+JavaScript Websocket SDK.
+
+### Connecting to Pusher
+
+Client connections require an application key to be provided. If the client will be subscribing
+to private or presence channels, a server-side authentication endpoint must be provided as well.
+The client attempts to use `/pusher/auth` path for the endpoint if one is not explicitly provided.
+
+The following example illustrates how to instantiate a Pusher connection using a custom authentication
+ endpoint. For more information about connections, refer to
+ [this page](https://pusher.com/docs/client_api_guide/client_connect).
```

Comment:
"For more information about connections, refer to [this page](https://pusher.com/docs/client_api_guide/client_connect)."

Change to:

"For more information about connections, refer to Pusher's [Connection](https://pusher.com/docs/client_api_guide/client_connect) documentation."